src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -1097,7 +1097,8 @@ public void visitSource(String sourceName, SCMSourceObserver observer)
                 SourceFactory sourceFactory = new SourceFactory(request);
                 WitnessImpl witness = new WitnessImpl(listener);
 
-                if (!github.isAnonymous()) {
+                boolean githubAppAuthentication = credentials instanceof GitHubAppCredentials;
+                if (!github.isAnonymous() && !githubAppAuthentication) {
```

Comment:
The logging is important. Users need this information when it happens.
I understand the other code path doesn't log, but that doesn't mean it is correct either. 

Suggest making this change to both this and L936:

```suggestion
                if (github.isAnonymous()) {
                    listener.getLogger().format("Connecting to %s with no credentials, anonymous access%n",
                            apiUri == null ? GitHubSCMSource.GITHUB_URL : apiUri);                
                } else if (!githubAppAuthentication) {
```